gen/decoder.go

```diff
@@ -257,14 +257,31 @@ func (g *Generator) genTypeDecoderNoCheck(t reflect.Type, out string, tags field
 
 }
 
-func (g *Generator) genStructFieldDecoder(t reflect.Type, f reflect.StructField) error {
+func (g *Generator) genStructFieldDecoder(t reflect.Type, f reflect.StructField, inline bool) error {
 	jsonName := g.fieldNamer.GetJSONFieldName(t, f)
 	tags := parseFieldTags(f)
 
 	if tags.omit {
 		return nil
 	}
 
+	if inline {
+		if tags.inline {
+			// each inline should clone Lexer Data
+			fmt.Fprintln(g.out, "{")
+			fmt.Fprintln(g.out, "  in := &jlexer.Lexer{Data: in.Data}")
```

Comment:
You are making a fresh lexer with all the data, but without the position. Thus, you are reading from the beginning of the entire JSON document.